storage/filestore.go

```diff
@@ -0,0 +1,205 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/docker/notary"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// NewFilesystemStore creates a new store in a directory tree
+func NewFilesystemStore(baseDir, subDir, extension string) (*FilesystemStore, error) {
+	baseDir = path.Join(baseDir, subDir)
+
+	return NewFileStore(baseDir, extension, notary.PrivKeyPerms)
+}
+
+// NewFileStore creates a fully configurable file store
+func NewFileStore(baseDir, fileExt string, perms os.FileMode) (*FilesystemStore, error) {
+	baseDir = filepath.Clean(baseDir)
+	if err := createDirectory(baseDir, perms); err != nil {
```

Comment:
Non-blocking: Do we need to create a directory when we create a new `FilesystemStore`?  Set will create all of them when adding a file.

Is it just that we want to make sure the directory doesn't already exist as a file or something?